src/fsharp/CompileOps.fs

```diff
@@ -4519,17 +4518,19 @@ type TcImports(tcConfigP:TcConfigProvider, initialResolutions:TcAssemblyResoluti
 
 /// Process #r in F# Interactive.
 /// Adds the reference to the tcImports and add the ccu to the type checking environment.
-let RequireDLL (tcImports:TcImports) tcEnv m file = 
-    let RequireResolved = function
-        | ResolvedImportedAssembly(ccuinfo) -> ccuinfo
-        | UnresolvedImportedAssembly(assemblyName) -> error(Error(FSComp.SR.buildCouldNotResolveAssemblyRequiredByFile(assemblyName,file),m))
+let RequireDLL (tcImports:TcImports, tcEnv, thisAssemblyName, m, file) = 
     let resolutions = CommitOperationResult(tcImports.TryResolveAssemblyReference(AssemblyReference(m,file,None),ResolveAssemblyReferenceMode.ReportErrors))
     let dllinfos,ccuinfos = tcImports.RegisterAndImportReferencedAssemblies(resolutions)
-    let ccuinfos = ccuinfos |> List.map RequireResolved
+   
+    let asms = 
+        ccuinfos |> List.map  (function
+            | ResolvedImportedAssembly(asm) -> asm
+            | UnresolvedImportedAssembly(assemblyName) -> error(Error(FSComp.SR.buildCouldNotResolveAssemblyRequiredByFile(assemblyName,file),m)))
+
     let g = tcImports.GetTcGlobals()
     let amap = tcImports.GetImportMap()
-    let tcEnv = ccuinfos |> List.fold (fun tcEnv ccuinfo -> Tc.AddCcuToTcEnv(g,amap,m,tcEnv,ccuinfo.FSharpViewOfMetadata,ccuinfo.AssemblyAutoOpenAttributes,false)) tcEnv 
-    tcEnv,(dllinfos,ccuinfos)
+    let tcEnv = (tcEnv, asms) ||> List.fold (fun tcEnv asm -> Tc.AddCcuToTcEnv(g,amap,m,tcEnv,thisAssemblyName,asm.FSharpViewOfMetadata,asm.AssemblyAutoOpenAttributes,asm.AssemblyInternalsVisibleToAttributes)) 
```

Comment:
This correctly propagates IVT attributes down for ``#r`` assemblies.  I noticed it when doing unit testing on assemblies using F# Interactive.